src/emqx_banned.erl

```diff
@@ -159,3 +160,10 @@ expire_banned_items(Now) ->
          (_, _Acc) -> ok
       end, ok, ?BANNED_TAB).
 
+run_hooks(Banned) ->
+    case emqx:get_env(clean_when_banned) of
+        true ->
+            emqx_hooks:run('client.banned', [Banned]);
```

Comment:
this config check should be done inside the hook callback ?

Reply:
But that may have some performance overheads